primer/index.md

```diff
@@ -2024,561 +2025,194 @@ void read_from_file(){
-  boolean            -        bool            O
-```
-
-***Processing N-tuples Spanning over Several Files***
-
-Usually n-tuples or trees span over many files and it would be difficult
-to add them manually. ROOT thus kindly provides a helper class in the
-form of `TChain`. Its usage is shown in the following macro which is
-very similar to the previous example. The constructor of a `TChain`
-takes the name of the `TTree` (or `TNuple`) as an argument. The files
-are added with the function `Add(fileName)`, where one can also use
-wild-cards as shown in the example.
+Or maybe you can try producing a plot out of a newly defined variable starting
+from the available columns:
 
 {% highlight C++ %}
-// Read several previously produced N-Tuples and print on screen its
-// content.
-//
-// you can easily create some files with the following statement:
-//
-// for i in 0 1 2 3 4 5; \\
-// do root -l -x -b -q \\
-// "write_ntuple_to_file.cxx \\
-// (\"conductivity_experiment_${i}.root\", 100)"; \\
-//  done
-
-void read_ntuple_with_chain(){
-   // initiate a TChain with the name of the TTree to be processed
-   TChain in_chain("cond_data");
-   in_chain.Add("conductivity_experiment*.root"); // add files,
-                                                  // wildcards work
-
-   // define variables and assign them to the corresponding branches
-   float pot, cur, temp, pres;
-   in_chain.SetBranchAddress("Potential", &pot);
-   in_chain.SetBranchAddress("Current", &cur);
-   in_chain.SetBranchAddress("Temperature", &temp);
-   in_chain.SetBranchAddress("Pressure", &pres);
-
-   cout << "Potential\tCurrent\tTemperature\tPressure\n";
-   for (size_t irow=0; irow<in_chain.GetEntries(); ++irow){
-      in_chain.GetEntry(irow); // loads all variables that have
-                                    // been connected to branches
-      cout << pot << "\t" << cur << "\t" << temp <<
-                          "\t" << pres << endl;
-   }
-}
+root [5] auto g3 = df.Define("curoverpot", "current/potential").Graph("curoverpot","temperature");
+root [6] g3->Draw("ap");
 {% endhighlight %}
 
-***For the advanced user: Processing trees with a selector script***
-
-
-Another very general and powerful way of processing a `TChain` is
-provided via the method `TChain::Process()`. This method takes as
-arguments an instance of a -- user-implemented-- class of type
-`TSelector`, and -- optionally -- the number of entries and the first
-entry to be processed. A template for the class `TSelector` is provided
-by the method `TTree::MakeSelector`, as is shown in the little macro
-`makeSelector.C` below.
+## Working with columns that hold collections
 
-It opens the n-tuple `conductivity_experiment.root` from the example
-above and creates from it the header file `MySelector.h` and a template
-to insert your own analysis code, `MySelector.C`.
+RDataFrame reads collections as the special type
+[ROOT::RVec](https://root.cern/doc/master/classROOT_1_1VecOps_1_1RVec.html):
```

Comment:
```suggestion
[ROOT::VecOps::RVec](https://root.cern/doc/master/classROOT_1_1VecOps_1_1RVec.html):
```

Reply:
I'd rather advertise it as `ROOT::RVec` here (it's valid)